iaso/api/serializers.py

```diff
@@ -159,7 +159,13 @@ class Meta:
 class OrgUnitSearchParentSerializer(OrgUnitSerializer):
     class Meta:
         model = OrgUnit
-        fields = ["id", "name", "parent"]
+        fields = [
+            "id",
+            "name",
+            "parent",
+            "org_unit_type_id",
+            "org_unit_type_name",
```

Comment:
We need a prefetch_related on tha field

Reply:
@beygorghor I think what we really need is a `select_related()`, not a `prefetch_related()`, because [`"org_unit_type_name"`](https://github.com/BLSQ/iaso/blob/38ea50276cd283c99623d9ceeca530e4a75c3dd3/iaso/api/serializers.py#L167) is calling [`def get_org_unit_type_name()`](https://github.com/BLSQ/iaso/blob/38ea50276cd283c99623d9ceeca530e4a75c3dd3/iaso/api/serializers.py#L84-L85) and it's a foreign key.

We need to optimize the view where [`OrgUnitSearchSerializer` is called](https://github.com/BLSQ/iaso/blob/c9f6f24001d7d2648f284d197bca4c63024ac391/iaso/api/org_units.py#L156).

The only problem is that this view is a bit of a mess in terms of where the query is performed…

So we could [replace this line](https://github.com/BLSQ/iaso/blob/c9f6f24001d7d2648f284d197bca4c63024ac391/iaso/api/org_units.py#L141):

```python
queryset = queryset.order_by(*order)
```

by:

```python
queryset = queryset.select_related("parent__org_unit_type").order_by(*order)
```

Reply:
@beygorghor this wasn't the right place, my bad. I fixed it here https://github.com/BLSQ/iaso/pull/1117/commits/0464a4867ad98ffeaf957f727ec86d7996ac7b7e

Don't forget to pull.